src/registry.js

```diff
@@ -59,7 +60,7 @@ class CommandoRegistry {
 	registerGroup(group, name, guarded) {
 		if(typeof group === 'string') {
 			group = new CommandGroup(this.client, group, name, guarded);
-		} else if(typeof group === 'function') {
+		} else if(isConstructor(group, CommandGroup)) {
 			group = new group(this.client); // eslint-disable-line new-cap
 		} else if(typeof group === 'object' && !(group instanceof CommandGroup)) {
 			group = new CommandGroup(this.client, group.id, group.name, group.guarded);
```

Comment:
After my merge conflict resolution I noticed this doesn't follow the new pattern of `x || x.default` being a constructor. Is this intentional?

```suggestion
		} else if(isConstructor(group, CommandGroup)) {
			group = new group(this.client); // eslint-disable-line new-cap
		} else if(isConstructor(group.default, CommandGroup)) {
			group = new group.default(this.client); // eslint-disable-line new-cap
		} else if(typeof group.default === 'object' && !(group.default instanceof CommandGroup)) {
			group = new CommandGroup(this.client, group.default.id, group.default.name, group.default.guarded);
		} else if(typeof group === 'object' && !(group instanceof CommandGroup)) {
			group = new CommandGroup(this.client, group.id, group.name, group.guarded);
```